src/code/z_camera.c

```diff
@@ -3106,266 +6797,795 @@ void func_80057C6C(Camera* camera, View* view, CollisionContext* colCtx, GlobalC
+        Math_Coss(D_8011DB08);
+        Math_Sins(D_8011DB08);
+        Math_Sins(D_8011DB0C);
+        func_800AA76C(&camera->globalCtx->view, 0.0f, 0.0f, 0.0f);
+        func_800AA78C(&camera->globalCtx->view, Math_Sins(D_8011DB0C) * (sp40 * phi_f20) + 1.0f,
+                      Math_Coss(D_8011DB0C) * (sp3C * phi_f20) + 1.0f, Math_Coss(D_8011DB08) * (sp38 * phi_f20) + 1.0f);
+        func_800AA7AC(&camera->globalCtx->view, sp34 * sp60);
+        camera->unk_14C |= 0x40;
+    } else if (camera->unk_14C & 0x40) {
+        func_800AA814(&camera->globalCtx->view);
+        camera->unk_14C &= ~0x40;
+    }
+}
+
+Vec3s* Camera_Update(Vec3s* outVec, Camera* camera) {
+    static s32 sOOBTimer = 0;
+
+    Vec3f viewAt;
+    Vec3f viewEye;
+    Vec3f viewUp;
+    f32 viewFov;
+    Vec3f spAC;
+    u32 bgCheckId;
+    f32 playerGroundY;
+    f32 playerXZSpeed;
+    VecSph eyeAtAngle;
+    s16 camDataIdx;
+    PosRot curPlayerPosRot;
+    QuakeCamCalc quake;
+    Player* player;
+
+    player = camera->globalCtx->cameraPtrs[0]->player;
+
+    if (R_DBG_CAM_UPDATE) {
+        osSyncPrintf("camera: in %x\n", camera);
+    }
+
+    if (camera->status == CAM_STAT_CUT) {
+        if (R_DBG_CAM_UPDATE) {
+            osSyncPrintf("camera: cut out %x\n", camera);
+        }
+        *outVec = camera->inputDir;
+        return outVec;
+    }
+
+    sUpdateCameraDirection = false;
+
+    if (camera->player != NULL) {
+        func_8002EF44(&curPlayerPosRot, &camera->player->actor);
+        camera->xzSpeed = playerXZSpeed = OLib_Vec3fDistXZ(&curPlayerPosRot.pos, &camera->playerPosRot.pos);
+
+        camera->speedRatio = OLib_ClampMaxDist(playerXZSpeed / (func_8002DCE4(camera->player) * PCT(OREG(8))), 1.0f);
+        camera->playerPosDelta.x = curPlayerPosRot.pos.x - camera->playerPosRot.pos.x;
+        camera->playerPosDelta.y = curPlayerPosRot.pos.y - camera->playerPosRot.pos.y;
+        camera->playerPosDelta.z = curPlayerPosRot.pos.z - camera->playerPosRot.pos.z;
+        spAC = curPlayerPosRot.pos;
+        spAC.y += Player_GetHeight(camera->player);
+
+        playerGroundY = func_8003CA0C(camera->globalCtx, &camera->globalCtx->colCtx, &playerFloorPoly, &bgCheckId,
+                                      &camera->player->actor, &spAC);
+        if (playerGroundY != BGCHECK_Y_MIN) {
+            // player is above ground.
+            sOOBTimer = 0;
+            camera->floorNorm.x = playerFloorPoly->norm.x * COLPOLY_NORM_FRAC;
+            camera->floorNorm.y = playerFloorPoly->norm.y * COLPOLY_NORM_FRAC;
+            camera->floorNorm.z = playerFloorPoly->norm.z * COLPOLY_NORM_FRAC;
+            camera->bgCheckId = bgCheckId;
+            camera->playerGroundY = playerGroundY;
+        } else {
+            // player is not above ground.
+            sOOBTimer++;
+            if (1) {
+                camera->floorNorm.x = 0.0;
+                camera->floorNorm.y = 1.0f;
+                camera->floorNorm.z = 0.0;
             }
```

Comment:
can the if 1 be on its own line or does it have to wrap this stuff?

Reply:
I was actually able to remove the if altogether. FeelsGoodMan